specs/rollup-node.md

```diff
@@ -251,6 +253,46 @@ However, they can still be challenged by a fault proof until the end of the faul
 
 > **TODO** the spec doesn't encode the notion of fault proof yet, revisit this (and include links) when it does
 
+## From L2 Block to L2 Output Root
+
+After processing a block the resulting outputs will need to be synchronized with L1 for trustless execution of
+L2-to-L1 messaging, such as withdrawals. To synchronize outputs are merkleized: hashed in a structured form for minimal
+proof cost to any piece of data. The merkle-structure is defined with [SSZ], a type system for merkleization and
+serialization, used in L1 (beacon-chain). However, we replace `sha256` with `keccak256` to save gas costs in the EVM.
+
+[SSZ]: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md
+
+```python
+class L2Output(Container):
+  state_root: Bytes32
+  withdrawal_storage_root: Bytes32
+  latest_block: ExecutionPayload  # includes block-hash
+  history_accumulator_root: Bytes32  # Not functional yet
+  extension: Bytes32
+```
+
+The `state_root` is the Merkle-Patricia-Trie ([MPT][g-mpt]) root of all execution-layer accounts,
+also found in `latest_block.state_root`: this field is frequently used and thus elevated closer to the L2 output root,
+reducing the merkle proof depth and thus the cost of usage.
+
+The `withdrawal_storage_root` elevates the Merkle-Patricia-Trie ([MPT][g-mpt]) root of L2 Withdrawal contract storage.
+Instead of a MPT proof to the Withdrawal contract account in the account trie,
+one can directly access the MPT storage trie root, thus reducing the verification cost of withdrawals on L1.
```

Comment:
I don't think we've 100% committed to doing this work yet.

Reply:
https://github.com/ethereum-optimism/optimistic-specs/pull/182#issuecomment-1029296578 explains the reasoning, we can further optimize later for the last 10%, and the current change is fully backwards compatible (with pre 1.0 contracts).